### PR TITLE
fix(linux): Fix display of installed keyboard version

### PR DESCRIPTION
**Pull request by ermshiperete**
Previously if there was a newer version available on keyman.com we showed that version number instead of the version number of the installed keyboard.

# User Testing

## Preparations

- The tests can be run on any Linux version
- [Install build artifacts of this PR](https://github.com/keymanapp/keyman/wiki/How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux)
- Reboot
- Download EuroLatin (SIL) version 2.0.1: <https://keyman.com/go/package/download/sil_euro_latin?version=2.0.1&tier=stable>
- Open Keyman Configuration
- If you have any EuroLatin (SIL) keyboards installed, uninstall them
- Install the just downloaded keyboard by clicking the "Install keyboard" button and selecting the downloaded file

## Tests

**TEST_VERSION_NO**: Correct version number shows

- Verify that the "Version" column shows version 2.0.1 for the EuroLatin (SIL) keyboard

### CODE REVIEW

File: linux/keyman-config/keyman_config/kmpmetadata.py
Comment (mcdurdin) [inline linux/keyman-config/keyman_config/kmpmetadata.py:1]: Changes in this file don't appear to be related to the fix?

File: linux/keyman-config/keyman_config/list_installed_kmp.py [hunk 1]
```diff
@@ -122,32 +122,21 @@ def get_kmp_version(packageID):
...
+    version = _get_kmp_version_internal(packageID, InstallLocation.Shared, version)
+    version = _get_kmp_version_internal(packageID, InstallLocation.User, version)
     return version
 
 
+def _get_kmp_version_internal(packageId, location, previousVersion):
+    kmp = get_installed_kmp(location)
+    if packageId in kmp:
+        version = kmp[packageId]['kmpversion']
+        if previousVersion and previousVersion < version or not previousVersion:
```
Comment (mcdurdin) [inline linux/keyman-config/keyman_config/list_installed_kmp.py:135]: String comparison of version numbers is not safe: `'2.0.1' > '10.0'`.

Also, this if statement is kinda strange, mixing `and` and `or not` like that, perhaps could do instead (still need to fix the version comparison though):

```suggestion
        if not previousVersion or previousVersion < version:
```

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Need to fixup the version comparison first I think. I am not 100% sure where the fix for the PR title is, either :grin:

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_VERSION_NO (PASSED):** Tested with the attached PR build (17.0.120-1~PR-8948+lunar4) in Ubuntu 23.04 Lunar Linux OS (VM) and here is my observation: 1. Verified that the Version number appears correctly under the version column.

File: linux/keyman-config/keyman_config/list_installed_kmp.py (hunk 1 above)
Comment (ermshiperete) [reply linux/keyman-config/keyman_config/list_installed_kmp.py]: We changed the version number comparison a while ago, but looks this one slipped through...

File: linux/keyman-config/keyman_config/kmpmetadata.py
Comment (ermshiperete) [reply linux/keyman-config/keyman_config/kmpmetadata.py:1]: Right. I put the refactorings in a separate commit, but probably should have created a separate PR as well...

**Comment by ermshiperete** [comment]
> Need to fixup the version comparison first I think. I am not 100% sure where the fix for the PR title is, either grin

Maybe misleading. What I meant was that this change fixes the method that returns the version number. With that the correct version number will be displayed :smile:

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.123-alpha](https://keyman.com/downloads/releases/alpha/17.0.123)
